### PR TITLE
Update to net 8

### PR DESCRIPTION
**Pull request by shargon**

### CODE REVIEW

**Comment by cschuchardt88** [comment]
You need to change `main.yml` to `DOTNET_VERSION: 8.0.x`

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
   pull_request:
 
 env:
-  DOTNET_VERSION: 7.0.x
+  DOTNET_VERSION: 8.0.x
```
Comment (Jim8y) [inline .github/workflows/main.yml:9]: dev-container as well?

Comment (shargon) [reply .github/workflows/main.yml:9]: done

Comment (cschuchardt88) [reply .github/workflows/main.yml:9]: `global.json`

**Review by Jim8y — APPROVED** [review]

**Comment by shargon** [comment]
mmm... coverage 0%

**Comment by Jim8y** [comment]
> mmm... coverage 0%

Hahahahahahahhahahahhahahahhahhahahaha

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
       run: dotnet build ./src/Neo.Compiler.CSharp/Neo.Compiler.CSharp.csproj
     - name: Build Neo.SmartContract.Template and test templates
       run: |
-        dotnet pack ./src/Neo.SmartContract.Template/Neo.SmartContract.Template.csproj
+        dotnet pack --configuration Debug ./src/Neo.SmartContract.Template/Neo.SmartContract.Template.csproj
```
Comment (cschuchardt88) [inline .github/workflows/main.yml:32]:
```suggestion
        dotnet pack ./src/Neo.SmartContract.Template/Neo.SmartContract.Template.csproj --configuration Debug
```

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+  "image": "mcr.microsoft.com/devcontainers/dotnet:1-8.0-jammy",
+  "onCreateCommand": "./scripts/load_submodule.sh",
+  "postCreateCommand": "dotnet restore && dotnet build",
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "ms-dotnettools.csdevkit"
+      ]
+    }
+  }
```
Comment (cschuchardt88) [inline .devcontainer/devcontainer.json:14]: Can you please not change the spacing?

File: src/Neo.SmartContract.Analyzer/Neo.SmartContract.Analyzer.csproj [hunk 4]
```diff
@@ -20,12 +20,12 @@
...
-            <PrivateAssets>all</PrivateAssets>
-            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-        </PackageReference>
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.3.0"/>
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.3.0"/>
+		<PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.9.2" />
```
Comment (cschuchardt88) [inline src/Neo.SmartContract.Analyzer/Neo.SmartContract.Analyzer.csproj:27]: Change spacing back

File: tests/Directory.Build.props [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
-  </ItemGroup>
+	<ItemGroup>
+		<PackageReference Update="coverlet.collector" Version="6.0.2">
+			<PrivateAssets>all</PrivateAssets>
+			<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
+		<PackageReference Update="Microsoft.NET.Test.Sdk" Version="17.9.0" />
+		<PackageReference Update="MSTest.TestAdapter" Version="3.2.2" />
+		<PackageReference Update="MSTest.TestFramework" Version="3.2.2" />
+	</ItemGroup>
```
Comment (cschuchardt88) [inline tests/Directory.Build.props:17]: Change spacing back

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (shargon) [reply .devcontainer/devcontainer.json:14]: I didn't do nothing special, just my visual studio...

File: src/Neo.SmartContract.Analyzer/Neo.SmartContract.Analyzer.csproj (hunk 4 above)
Comment (shargon) [reply src/Neo.SmartContract.Analyzer/Neo.SmartContract.Analyzer.csproj:27]: visual studio format this file like that....

**Comment by shargon** [comment]
I don't know why it doesn't test anything... :S, @cschuchardt88 @Jim8y feel free to commit your changes

File: tests/Neo.SmartContract.Framework.TestContracts/Neo.SmartContract.Framework.TestContracts.csproj [hunk 6]
```diff
@@ -1,7 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>net7.0</TargetFramework>
```
Comment (AnnaShaleva) [inline tests/Neo.SmartContract.Framework.TestContracts/Neo.SmartContract.Framework.TestContracts.csproj:4]: Just a question: why don't we need a target framework for test contracts?

Comment (cschuchardt88) [reply tests/Neo.SmartContract.Framework.TestContracts/Neo.SmartContract.Framework.TestContracts.csproj:4]: The configurations are stored in `Directory.Build.props` file 
https://github.com/neo-project/neo-devpack-dotnet/blob/master/tests/Directory.Build.props

**Comment by superboyiii** [comment]
Can any of you guys successfully run the unit test on **windows**? I update MSBuild to v17.9.0, MSBuild Toolset to v17.9.0, .NET SDK to v8.0.3, but it still returns error that "The following test container was not found: 'E:\PR-test\devpack-PR#1004\superboy\neo-devpack-dotnet\tests\Neo.Compiler.CSharp.UnitTests\bin\Debug\net8.0\Neo.Compiler.CSharp.UnitTests.dll'" .
I'm running test cases by Visual Studio Visual Studio 2022 v17.9.4

**Comment by cschuchardt88** [comment]
> Can any of you guys successfully run the unit test on **windows**? I update MSBuild to v17.9.0, MSBuild Toolset to v17.9.0, .NET SDK to v8.0.3, but it still returns error that `"The following test container was not found: 'E:\PR-test\devpack-PR#1004\superboy\neo-devpack-dotnet\tests\Neo.Compiler.CSharp.UnitTests\bin\Debug\net8.0\Neo.Compiler.CSharp.UnitTests.dll'"` . I'm running test cases by Visual Studio Visual Studio 2022 v17.9.4

@shargon @superboyiii 
Set `global.json` settings to this `"rollForward": "disable",` and make sure you have `8.0.202` installed... `8.0.203` doesn't not have support for `Visual Studio`.

**Comment by superboyiii** [comment]
> `8.0.203` doesn't not have support for `Visual Studio`.

But why unit tests work on neo-modules? I run these by visual studio as well...
65850a5ea6f13d56f27615629ccd63b

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (shargon) [reply .devcontainer/devcontainer.json]: done

File: src/Neo.SmartContract.Analyzer/Neo.SmartContract.Analyzer.csproj (hunk 4 above)
Comment (shargon) [reply src/Neo.SmartContract.Analyzer/Neo.SmartContract.Analyzer.csproj]: done

File: tests/Directory.Build.props (hunk 5 above)
Comment (shargon) [reply tests/Directory.Build.props]: done

**Comment by shargon** [comment]
I don't know why we have coverage 0%

**Comment by shargon** [comment]
Solved

File: tests/Neo.SmartContract.Analyzer.UnitTests/Neo.SmartContract.Analyzer.UnitTests.csproj [hunk 7]
```diff
@@ -1,23 +1,23 @@
 <Project Sdk="Microsoft.NET.Sdk">
+    
+    <PropertyGroup>
+        <IsPackable>false</IsPackable>
+        <RootNamespace>Neo.SmartContract.Analyzer.UnitTests</RootNamespace>
```
Comment (cschuchardt88) [inline tests/Neo.SmartContract.Analyzer.UnitTests/Neo.SmartContract.Analyzer.UnitTests.csproj]:
```suggestion
        <IsTestProject>true</IsTestProject>
        <RootNamespace>Neo.SmartContract.Analyzer.UnitTests</RootNamespace>
```

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]

File: tests/Directory.Build.props [hunk 8]
```diff
@@ -2,15 +2,15 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <IsPackable>false</IsPackable>
```
Comment (shargon) [inline tests/Directory.Build.props:6]:
```suggestion
    <TargetFramework>net8.0</TargetFramework>
    <IsPackable>false</IsPackable>
    <IsTestProject>true</IsTestProject>
```

File: tests/Neo.SmartContract.Analyzer.UnitTests/Neo.SmartContract.Analyzer.UnitTests.csproj (hunk 7 above)
Comment (shargon) [reply tests/Neo.SmartContract.Analyzer.UnitTests/Neo.SmartContract.Analyzer.UnitTests.csproj]: Done in DirectoryBuilds

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -41,9 +41,9 @@ jobs:
...
-        dotnet ./src/Neo.Compiler.CSharp/bin/Debug/net7.0/nccs.dll ./src/Neo.SmartContract.Template/bin/Debug/ownable/Ownable.csproj -o ./tests/Neo.SmartContract.Template.UnitTests/templates/neocontractowner/Artifacts/
-        dotnet ./src/Neo.Compiler.CSharp/bin/Debug/net7.0/nccs.dll ./src/Neo.SmartContract.Template/bin/Debug/oracle/OracleRequest.csproj -o ./tests/Neo.SmartContract.Template.UnitTests/templates/neocontractoracle/Artifacts/
+        dotnet ./src/Neo.Compiler.CSharp/bin/Debug/net8.0/nccs.dll ./src/Neo.SmartContract.Template/bin/Debug/nep17/Nep17Contract.csproj -o ./tests/Neo.SmartContract.Template.UnitTests/templates/neocontractnep17/Artifacts/
+        dotnet ./src/Neo.Compiler.CSharp/bin/Debug/net8.0/nccs.dll ./src/Neo.SmartContract.Template/bin/Debug/ownable/Ownable.csproj -o ./tests/Neo.SmartContract.Template.UnitTests/templates/neocontractowner/Artifacts/
+        dotnet ./src/Neo.Compiler.CSharp/bin/Debug/net8.0/nccs.dll ./src/Neo.SmartContract.Template/bin/Debug/oracle/OracleRequest.csproj -o ./tests/Neo.SmartContract.Template.UnitTests/templates/neocontractoracle/Artifacts/
```
Comment (vncoelho) [inline .github/workflows/main.yml:46]: Maybe set the paths as variables

**Comment by Jim8y** [comment]
works on my windows machine

File: .github/workflows/main.yml (hunk 9 above)
Comment (shargon) [reply .github/workflows/main.yml:46]: It follow the previous file, improvements can be in a different pr.

**Review by Jim8y — DISMISSED** [review]

File: tests/Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj [hunk 10]
```diff
@@ -1,16 +1,18 @@
...
-  <ItemGroup>
-    <None Include="*.cs">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
-  </ItemGroup>
-  <ItemGroup>
-    <ProjectReference Include="..\..\src\Neo.SmartContract.Framework\Neo.SmartContract.Framework.csproj" />
-  </ItemGroup>
+    <PropertyGroup>
+        <ImplicitUsings>enable</ImplicitUsings>
```
Comment (Jim8y) [inline tests/Neo.Compiler.CSharp.TestContracts/Neo.Compiler.CSharp.TestContracts.csproj:4]:
```suggestion
```

**Review by Jim8y — APPROVED** [review]

**Pull request merged**